src/middlewared/middlewared/utils/threading.py

```diff
@@ -4,17 +4,31 @@
 import os
 import threading
 
+from truenas_pylibzfs import open_handle
+
 from .prctl import set_name
 
+thread_local_storage = threading.local()
 logger = logging.getLogger(__name__)
 counter = count(1)
-__all__ = ["set_thread_name", "start_daemon_thread", "IoThreadPoolExecutor", "io_thread_pool_executor"]
+__all__ = [
+    "set_thread_name",
+    "start_daemon_thread",
+    "IoThreadPoolExecutor",
+    "io_thread_pool_executor",
+    "thread_local_storage",
+]
 
 
 def set_thread_name(name):
     set_name(name)
 
 
+def initializer(thread_name, tls):
+    set_name(thread_name)
+    tls.lzh = open_handle()
```

Comment:
I presume this is cheap enough to open it in all new threads by default?

Reply:
Yep, it's cheap and memory efficient. Shouldn't be an issue.